Fix/Indication.cs

```diff
@@ -0,0 +1,133 @@
+            }
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.SenderCompID) is not Field senderField || string.IsNullOrEmpty(senderField.Value))
+            {
+                throw new ArgumentException("Message does not contain a SenderCompID");
+            }
+
+            SenderCompID = senderField.Value;
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.TargetCompID) is not Field targetField || string.IsNullOrEmpty(targetField.Value))
+            {
+                throw new ArgumentException("Message does not contain a TargetCompID");
+            }
+
+            TargetCompID = targetField.Value;
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.IOIID) is not Field IOIIDField || string.IsNullOrEmpty(IOIIDField.Value))
+            {
+                throw new ArgumentException("Message does not contain a IOIID");
+            }
+
+            IOIID = IOIIDField.Value;
+
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.IOITransType) is not Field IOITransTypeField)
+            {
+                throw new ArgumentException("Message does not contain a IOITransType");
+            }
+
+            IOITransType = (FieldValue?)IOITransTypeField;
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.Symbol) is Field symbolField && !string.IsNullOrEmpty(symbolField.Value))
+            {
+                Symbol = symbolField.Value;
+            }
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.Price) is Field priceField)
+            {
+                Price = (decimal?)priceField;
+            }
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.Side) is Field sideField)
+            {
+                Side = (FieldValue?)sideField;
+            }
+
+            if (message.Fields.Find(FIX_5_0SP2.Fields.IOIQty) is Field ioiQtyField && !string.IsNullOrEmpty(ioiQtyField.Value))
+            {
+                IOIQty = ioiQtyField.Value;
+            }
+
+            int index = -1;
+            for (int i = 0; i < message.Fields.Count; ++i)
+            {
+                if (message.Fields[i].Tag == FIX_5_0SP2.Fields.NoIOIQualifiers.Tag)
+                {
+                    index = i;
+                    break;
+                }
+            }
+
+            if (index >= 0)
+            {
+                var noGroups = (long?)message.Fields[index];
+                Qualifiers = new List<FieldValue>();
+                for (var i = 0; i < noGroups; ++i)
+                {
+                    var qualifier = (FieldValue?)message.Fields[index + 1 + i];
+                    if (qualifier != null)
+                        Qualifiers.Add(qualifier);
+                }
+            }
+
+            Messages = new List<Message>
+            {
+                message
+            };
+
+            Key = CreateKey(SenderCompID, TargetCompID, IOIID);
+        }
+
+        public List<Message> Messages { get; private set; }
+        public List<FieldValue> Qualifiers { get; private set; }

```

Comment:
Initialise these inline, they are non nullable so must be initialised but the constructor does not guarantee this. Just add '= new();' to these declarations and change the code that inits them to just add the new items to the collections.